railties/lib/rails/generators/rails/app/templates/app/controllers/application_controller.rb.tt

```diff
@@ -1,6 +1,10 @@
 class ApplicationController < ActionController::<%= options.api? ? "API" : "Base" %>
 <%- unless options.api? -%>
   # Only allow modern browsers supporting webp images, web push, badges, import maps, CSS nesting, and CSS :has.
+  # Note: When using a desktop browser's developer tools to simulate a phone, it may be identifying itself as an older browser which
+  # is not allowed by :modern. You can customize the user agent to meet the version requirements of :modern. Instructions for Chrome:
+  #   https://developer.chrome.com/docs/devtools/device-mode
+  # The required browser versions for :modern are defined in ActionController::AllowBrowser::BrowserBlocker::SETS
```

Comment:
I think this would fit better in the API docs for `allow_browser`, but probably not in the `ApplicationController` generated for all new apps

Reply:
I think that the whole range of what's allowed for allow_browser makes sense to have in the developer docs and clearly not stuff all that into a comment, however this honestly feels broken. This confusion isn't a hypothetical one, I just experienced it and linked to the other discussion online where others experienced it.

Maybe I wrote too much, but I do think some hint is warranted. This doesn't feel like a moment where I think, "Oh cool, I wonder what the other options are, let me check the docs." This instead felt like a moment of, "Something is clearly broken. I've already tried updating my browser and it's on latest" so I do not expect the docs to explain rails bugs, I expect them to explain supported configuration. (I get that it's not a bug, but that's the mindset one is in when encountering this because it's so confusing).

I totally respect the rails team to make this call. I just thought I'd attempt to propose a solution because I'm expecting this issue to get reported a bunch more times so we'll have to keep answering it.

Reply:
Can we just link to the documentation of that method here instead?

I'm not so sure about explaining how developer tools of browsers works. I'm all for pointing to which versions of browsers work, but teaching people how developers tools works is out of the scope of Rails.